packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -106,6 +125,20 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
     return null;
   }
 
+  double _getCollapsePadding(double t, _FlexibleSpaceBarSettings settings) {
+    switch (widget.collapseMode) {
+      case CollapseMode.pin:
+        return -(settings.maxExtent - settings.currentExtent);
+      case CollapseMode.none:
+        return 0.0;
+      case CollapseMode.parallax:
+        final double deltaExtent = settings.maxExtent - settings.minExtent;
+        return -new Tween<double>(begin: 0.0, end: deltaExtent / 4.0).lerp(t);
+      default:
+        throw new ArgumentError('Invalid collapse mode.');
```

Comment:
you should not have  default (it defeats the analyzer lint i mentioned before).
Just add `return null` after the switch.